### ISSUE TITLE
inference options discussion

### ISSUE THREAD

**Issue by rbavery** [SkyTruth/cerulean-cloud#27]
There are significant trade-offs between using torch scripting, torch tracing, or torch state_dict. differences between scripting and tracing defined here: https://paulbridger.com/posts/mastering-torchscript/

Finds from inspecting with @rodrigoalmeida94 
* pickling with fastai is not an option, it requires too many extra dependencies and is brittle. Ideally we only depend on torch, not fastai and not icevision for inference: https://docs.fast.ai/learner.html#Learner.export
* obscure error when trying to export with torch scripting. It's very possible the fastai unet learner violates one of these rules https://pytorch.org/docs/stable/jit_unsupported.html. This would require in the weeds mods that I don't think we want to focus on now if there are easier options (see below)
* saving a torch state_dict is no problem. Loading it however with only torch as a dependency is difficult because we'd need to pull out the fastai internals that construct the model in our inference code, create the base pytorch model class, and then load the state_dict into that model class. 
* So the simplest solution seems to be torch tracing to get inference running. However this requires us to:
    * fix the batch sized for inference as the one used for training (I think ideally we would use a batch size of 1, with the largest tile we can, and accumulate gradients on a larger step)
    *  only run inference on a single gpu since we train on a single gpu and torch tracing populates the traced model with the model parameters and specific name of the cuda device called when it is traced

I don't think the caveats mentioned above for torch tracing are a problem since it's ok if inference takes a while to run. torch tracing will make the best use of gpu resources it sounds like since the model gets jit compiled to machine code.
